src/iso_alloc_util.c

```diff
@@ -28,7 +28,7 @@ INTERNAL_HIDDEN INLINE int _iso_getcpu(void) {
     uintptr_t a;
     __asm__ volatile("mrs %x0, tpidrro_el0"
                      : "=r"(a)::"memory");
-    return (int) ((a & 0x8) - 1);
+    return (int) ((a & 0x8));
```

Comment:
This change should be part of another commit, since it's not really a cleanup, but a proper change.

Reply:
Agree. This slipped in. @devnexen this was the fix in my testing on an M1 macbook but it's not really tested or researched beyond that. Just FYI since you contributed this code.